src/deepsparse/pipeline.py

```diff
@@ -565,6 +566,71 @@ def _initialize_engine(self) -> Union[Engine, ORTEngine]:
             )
 
 
+class FunctionPipeline(Pipeline):
+    """
+    A utility class provided to make specifying custom pipelines easier.
+    Instead of creating a subclass of Pipeline, you can instantiate this directly
+    by passing in functions to call for pre and post processing.
+
+    For example, YOLO pipline could be specified as:
+    ```python
+    def yolo_preprocess(inputs: YOLOInput) -> List[np.ndarray]:
+        ...
+
+    def yolo_postprocess(engine_outputs: List[np.ndarray]) -> YOLOOutput:
+        ...
+
+    yolo = FunctionalPipeline(
+        model_path="...",
+        input_schema=YOLOInput,
+        output_schema=YOLOOutput,
+        process_inputs_fn=yolo_preprocess,
+        process_outputs_fn=yolo_postprocess,
+    )
+    ```
+    """
+
+    def __init__(
+        self,
+        model_path: str,
+        input_schema: Type[BaseModel],
+        output_schema: Type[BaseModel],
+        process_inputs_fn: Callable[
+            [BaseModel],
+            Union[List[numpy.ndarray], Tuple[List[numpy.ndarray], Dict[str, Any]]],
+        ],
+        process_outputs_fn: Callable[[List[numpy.ndarray]], BaseModel],
+        *args,
+        **kwargs,
+    ):
+        self._input_schema = input_schema
+        self._output_schema = output_schema
+        self._process_inputs_fn = process_inputs_fn
+        self._process_outputs_fn = process_outputs_fn
+        super().__init__(model_path, *args, **kwargs)
+
+    def setup_onnx_file_path(self) -> str:
+        return model_to_path(self.model_path)
+
+    @property
+    def input_schema(self) -> Type[BaseModel]:
+        return self._input_schema
+
+    @property
+    def output_schema(self) -> Type[BaseModel]:
+        return self._output_schema
+
+    def process_inputs(
+        self, inputs: BaseModel
+    ) -> Union[List[numpy.ndarray], Tuple[List[numpy.ndarray], Dict[str, Any]]]:
+        return self._process_inputs_fn(inputs)
+
+    def process_engine_outputs(
+        self, engine_outputs: List[numpy.ndarray], **kwargs
+    ) -> BaseModel:
```

Comment:
Do you prefer docstrings specific to this class or were oyu thinking just copy/paste Pipeline docstrings? Right now IDES will grab Pipeline docstrings for these methods since they don't have a new one specified

Reply:
if there's any wording that would change between the parent and this docstring i'd add a new one